examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java

```diff
@@ -0,0 +1,233 @@
+  }
+
+  public static void modelBigQueryIO(Pipeline p,
+      String writeProject, String writeDataset, String writeTable) {
+    {
+      // [START BigQueryTableSpec]
+      String tableSpec = "clouddataflow-readonly:samples.weather_stations";
+      // [END BigQueryTableSpec]
+    }
+
+    {
+      // [START BigQueryTableSpecWithoutProject]
+      String tableSpec = "samples.weather_stations";
+      // [END BigQueryTableSpecWithoutProject]
+    }
+
+    {
+      // [START BigQueryTableSpecObject]
+      TableReference tableSpec = new TableReference()
+          .setProjectId("clouddataflow-readonly")
+          .setDatasetId("samples")
+          .setTableId("weather_stations");
+      // [END BigQueryTableSpecObject]
+    }
+
+    {
+      String tableSpec = "clouddataflow-readonly:samples.weather_stations";
+      // [START BigQueryReadTable]
+      PCollection<Double> maxTemperatures = p
+          .apply(BigQueryIO.readTableRows().from(tableSpec))
+          // Each row is of type TableRow
+          .apply(MapElements.into(TypeDescriptors.doubles()).via(
+              (TableRow row) -> (Double) row.get("max_temperature")));
+      // [END BigQueryReadTable]
+    }
+
+    {
+      // [START BigQueryReadQuery]
+      PCollection<Double> maxTemperatures = p
+          .apply(BigQueryIO.readTableRows().fromQuery(
+              "SELECT max_temperature FROM [clouddataflow-readonly:samples.weather_stations]"))
+          // Each row is of type TableRow
+          .apply(MapElements.into(TypeDescriptors.doubles()).via(
+              (TableRow row) -> (Double) row.get("max_temperature")));
+      // [END BigQueryReadQuery]
+    }
+
+    {
+      // [START BigQueryReadQueryStdSQL]
+      PCollection<Double> maxTemperatures = p
+          .apply(BigQueryIO.readTableRows().fromQuery(
+              "SELECT max_temperature FROM `clouddataflow-readonly.samples.weather_stations`")
+              .usingStandardSql())
+          // Each row is of type TableRow
+          .apply(MapElements.into(TypeDescriptors.doubles()).via(
+              (TableRow row) -> (Double) row.get("max_temperature")));
+      // [END BigQueryReadQueryStdSQL]
+    }
+
+    {
+      // [START BigQuerySchema]
+      String tableSchema = "source:STRING, quote:STRING";
+      // [END BigQuerySchema]
+    }
+
+    // [START BigQuerySchemaObject]
+    List<TableFieldSchema> fields = new ArrayList<>(Arrays.asList(
+        new TableFieldSchema().setName("source").setType("STRING").setMode("NULLABLE"),
+        new TableFieldSchema().setName("quote").setType("STRING").setMode("REQUIRED")));
+    TableSchema tableSchema = new TableSchema().setFields(fields);
+    // [END BigQuerySchemaObject]
+
+    {
+      String tableSpec = "clouddataflow-readonly:samples.weather_stations";
+      if (writeProject != "" && writeDataset != "" && writeTable != "") {
+        tableSpec = writeProject + ":" + writeDataset + "." + writeTable;
+      }
+
+      // [START BigQueryWrite]
```

Comment:
<!--new_thread; commit:9eba19b98f4e946b4d764bd60be0b2a39d3a8124; resolved:0-->
Do we need the commented part?

Reply:
It's not needed, but as part of a snippet, it's good for the reader to know the whole context :)